comp/collector/collector/collectorimpl/collector_demux_test.go

```diff
@@ -90,14 +130,16 @@ func (suite *CollectorDemuxTestSuite) TestCancelledCheckCanSendMetrics() {
 
 	flop <- struct{}{}
 
-	suite.waitForCancelledCheckMetrics()
+	suite.waitForCheckMetrics()
+	id := <-suite.SenderManagerMock.destroyChannel
```

Comment:
Should we have `waitForCheckMetrics` reading `destroyChannel` and returning the id (instead of a two separate calls) or is it impossible/impacting other tests?

Reply:
I this would be possible without impacting other tests, but I'm not quite sure what the reason for that would be, would you mind explaining?

Reply:
To be clear, do you mean something like this:
`id := suite.waitForCheckMetrics(<-suite.SenderManagerMock.destroyChannel)`